docs/config/node.rst

```diff
@@ -851,15 +851,19 @@ Queries
   | *Default:* ``8192``
   | *Runtime:* ``no``
 
-  This setting defines the maximum number of elements an array can have so that
-  the ``!= ANY()``, ``LIKE ANY()``, ``ILIKE ANY()``, ``NOT LIKE ANY()`` and the
-  ``NOT ILIKE ANY()`` :ref:`operators <gloss-operator>` can be applied on it.
+  This setting limits the number of boolean clauses that can be generated by
+  ``!= ANY()``, ``LIKE ANY()``, ``ILIKE ANY()``, ``NOT LIKE ANY()`` and
+  ``NOT ILIKE ANY()`` :ref:`operators <gloss-operator>` on arrays in order to
+  prevent executing queries that may result in heavy memory consumptions
+  causing nodes to crash with ``OutOfMemory`` exceptions. Throws
+  ``TooManyClauses`` exceptions when exceeded.
 
   .. NOTE::
 
-    Increasing this value to a large number (e.g. 10M) and applying  those
-    ``ANY`` operators on arrays of that length can lead to heavy memory,
-    consumption which could cause nodes to crash with OutOfMemory exceptions.
+    Resolve ``TooManyClauses`` exceptions by increasing this setting.
+    Guesstimate the value starting from the fact that
```

Comment:
I wouldn't mention anything about a formula, I think something like this:
`the number of boolean clauses used can be larger than the elements of the array`
is enough.